extension/event-page.js

```diff
@@ -93,32 +97,55 @@ function updateBadge(tabId, response) {
 }
 
 function filterResponse(response) {
-
   if (!response) return response;
 
-  let ret = response.filter(entity => /^https?:\/\/schema.org\/(Product|Event)$/.test(entity['@type']));
+  let ret = response.filter(entity => {
+    if (/^https?:\/\/schema.org\/(Product|Event)$/.test(entity['@type'])) return true;
+    if (entity['@type'].endsWith('CustomEvent')) return true;
+    if (entity['@type'] == manifestType) return true;
+    return false;
+  });
   return ret;
 }
 
-function updateArc(tabId, response) {
+function loadManifest(url) {
+  return loader.loadResource(url).then(contents => {
+    path = url.split('/').slice(0, -1).join('/');
+    return Arcs.Manifest.parse(contents,
+      {fileName: url, //url.split('/').slice(-1)[0],
+        path: path,
+        loader: new Arcs.BrowserLoader(path)
+      }
+    )}
+  );
+}
 
+function updateArc(tabId, response) {
   if (!response) return;
   console.log('updating tab '+tabId+' with response', response);
 
   let entities = {};
 
+  additionalManifestUrls = response.filter(r => r['@type']==manifestType).map(r => r['url']);
+  additionalManifests = additionalManifestUrls.map(url => loadManifest(url));
 
-  manifest.then(m => {
-    response.forEach(r => {
+  Promise.all(additionalManifests).then(manifests => {
```

Comment:
Add TODO to build unit tests for this since it's likely fragile and may be source of current issue with injected manifests?

Reply:
Good call - this code is deleted (the logic moved) but I'll add a comment.